docs/styleguide.md

```diff
@@ -0,0 +1,198 @@
+    * `BBCLASSEXTEND`
+
+</details>
+
+Contributors are free to use a single newline to create logical breaks between dissimilar variables *within* a section, and are encouraged to use double-newlines between each section.
+
+
+## Variables
+
+### Metadata
+
+**BUGTRACKER**
+* Set to the URL of the upstream project's bugtracker.
+    * If the recipe source lives in the meta-nilrt layer directly, do not set this variable. Users can infer that the bugtracker is the `HOMEPAGE` - which should be set to the meta-nilrt upstream.
+
+**DESCRIPTION**
+* can be multiline.
+* should describe the package in complete sentences.
+* Prefer the expanded "NI LinuxRT" form, to maximize users' understanding.
+
+**HOMEPAGE**
+* Set to the URL of the project upstream.
+    * If the recipe source lives in the meta-nilrt layer directly, set the homepage to the meta-nilrt canonical GH repo. `HOMEPAGE = "https://github.com/ni/meta-nilrt"`
+
+**SUMMARY**
+* Limit SUMMARY to 80 characters.
+* Prefer the abbreviated "NILRT" form, to conserve characters.
+
+
+# .patch Files
+
+[Example of a good .patch file commit.](https://github.com/ni/meta-nilrt/pull/50/commits/73b046c57d73e188a3bf4adbf0965aa9312ebe08)
+
+`.patch` files should include the original author's commit message and meta information at their top. Information about the OE context (like why the `.patch` file is needed for the recipe) should be added after the original author's commit.
+
+At a minimum, you should add an additional message trailer declaring the [upstream status](https://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines#Patch_Header_Recommendations:_Upstream-Status) of the .patch file at the time of your OE commit. These status lines are crucial to helping the project maintainers properly upgrade and rebase recipes. Common status lines are:
+
+* `Upstream-Status: Inappropriate [$rationale]` ([ex](https://github.com/ni/meta-nilrt/blob/nilrt/master/sumo/recipes-support/curl/curl/0005-Add-nicurl-wrapper-functions.patch)) For when the `.patch` change is specific to NI LinuxRT and would not be desired (or has been rejected) by upstream.
+* `Upstream-Status: Not Submitted [$rationale]` ([ex](https://github.com/ni/meta-nilrt/blob/nilrt/master/sumo/recipes-gnome/florence/files/0004-Add-option-for-automatic-bring-to-top.patch)) For when the `.patch` file is being included in NILRT before being submitted to its upstream project. This should only occur if the `.patch` is needed for an immediate NILRT release and there is no time to get it reviewed upstream beforehand. Or if - as in the case of the example - the upstream mailing list is dead.
+* `Upstream-Status: Submitted [$upstream_mailing_list_link]` ([ex](https://github.com/ni/meta-nilrt/blob/1e6453ebca8735de96eaaf3bc931d22998c8dfb3/recipes-support/curl/curl/0014-Fixup-lib1529-test.patch)) For when the `.patch` has been submitted to the upstream project's mailing list, but needs to be pulled into NILRT prior to final upstream approval.
```

Comment:
nit: some upstream projects don't use mailing lists, patch submissions happen via a web PR system. Would we have (for example) a GitHub or GitLab PR URL as the "upstream mailing list link" in that situation?

Reply:
Yes. I'll add some clarification to the doc.